Code/Framework/AzToolsFramework/AzToolsFramework/Entity/EditorEntityModel.cpp

```diff
@@ -381,12 +381,22 @@ namespace AzToolsFramework
             return;
         }
 
-        //orphan any children that remain attached to the entity
-        auto children = entityInfo.GetChildren();
-        for (auto childId : children)
-        {
-            ReparentChild(childId, AZ::EntityId(), entityId);
-            m_entityOrphanTable[entityId].insert(childId);
+        bool isPrefabSystemEnabled = false;
+        AzFramework::ApplicationRequests::Bus::BroadcastResult(
+            isPrefabSystemEnabled, &AzFramework::ApplicationRequests::IsPrefabSystemEnabled);
+
+        // For slices, orphan any children that remain attached to the entity
+        // For prefabs, this is an unneeded operation because the prefab system handles the orphans
+        // and the extra reparenting operation can be problematic for consumers subscribed to entity
+        // events, such as the entity outliner.
```

Comment:
nit: Just say 'Prefabs will handle any orphans on their own'.

When deleting an entity in prefab mode, it's currently not possible to orphan its children. If an entity is selected to delete, all its children are deleted too. I can't think of the exact reason this decision was made. Probably to simplify the workflows. If the user wants to not delete the children, they can reparent them to the grandparent entity or any other entity before deleting the parent. It's one additional action per delete. But given the issues we ran into with this, I think it's a worthy sacrifice. Besides, if users really want an option to orphan entities to grandparent, that can be something that's added later by prefabs as a separate workflow rather than do it in the outliner model like this.